test/unit/test_mechcat.cpp

```diff
@@ -282,10 +282,8 @@ TEST(mechcat, names) {
 TEST(mechcat, loading) {
     EXPECT_THROW(load_catalogue(LIBDIR "/does-not-exist-catalogue.so"), file_not_found_error);
     EXPECT_THROW(load_catalogue(LIBDIR "/libarbor.a"), bad_catalogue_error);
-    const mechanism_catalogue* cat = nullptr;
-    EXPECT_NO_THROW(cat = &load_catalogue(LIBDIR "/dummy-catalogue.so"));
-    ASSERT_NE(cat, nullptr);
-    EXPECT_EQ(std::vector<std::string>{"dummy"}, cat->mechanism_names());
+    const mechanism_catalogue cat = load_catalogue(LIBDIR "/dummy-catalogue.so");
```

Comment:
FYI, I don't think these tests are run by default on our CI.
When building with meson, the dynamic libraries have the extension `.dylib`, not `.so`, which causes this to break.
Not something to fix here (a bit out of scope), but something that will bite us on the behind at some point.

Reply:
In the CMake build they are renamed to avoid this exactl problem. Is that an option in Meson?

Reply:
In Meson this is the `name_suffix: 'so'` kwarg.

That being said, I would think for OS integration it might feel a bit more natural to use something like `LIBDIR "/dummy-catalogue." LIB_EXT` and pass that as a per-platform define?